File: frontend/src/types/stores/Team.ts
// src/types/stores/Team.ts
import { playerBasic } from './Player';

export interface TeamState {
  allTeams: Array<teams>;
  singleTeam: singleTeam;
}

export interface teams {
  conferenceName: string;
  divisionName: string;
  teamId: string;
  teamEnName: string;
  teamTwName: string;
  teamColor: string;
}

export interface loadSingleTeamPayload {
  id: string;
}

export interface singleTeam {
  roster: Array<playerBasic>;
}

// export interface loadSingleTeamParams {
//   id: string,
// }

// export interface loadTeamData2Params {
//   req1: string,
//   req2: number,
// }

File: backend/internal/model/team.go
// internal/model/team.go
package model

type Team struct {
	ConferenceName string `db:"conference_name" json:"conferenceName"`
	DivisionName   string `db:"division_name" json:"divisionName"`
	TeamID         string `db:"team_id" json:"teamId"`
	TeamENName     string `db:"team_en_name" json:"teamEnName"`
	TeamTWName     string `db:"team_tw_name" json:"teamTwName"`
	TeamColor      string `db:"team_color" json:"teamColor"`
}

type SingleTeam struct {
	Roster []PlayerBasic `db:"roster" json:"roster"`
}
